From 9f8b9e8733f123afd7589314cdb6afc9944b19ae Mon Sep 17 00:00:00 2001
From: Monk Liu <Monk.Liu@amd.com>
Date: Tue, 8 Aug 2017 16:16:29 +0800
Subject: [PATCH 42/46] amdgpu:fix potential deadlock

deadlock could occure between cpu mutex lock and
bo table mutex lock, this patch avoid it.

Change-Id: I083e402dde48f02a8ee196e59aa0cab80849fc18
Signed-off-by: Monk Liu <Monk.Liu@amd.com>
Acked-by: Qiang Yu <Qiang.Yu@amd.com>
---
 amdgpu/amdgpu_bo.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/amdgpu/amdgpu_bo.c b/amdgpu/amdgpu_bo.c
index 09028c7..c973a0d 100644
--- a/amdgpu/amdgpu_bo.c
+++ b/amdgpu/amdgpu_bo.c
@@ -673,10 +673,10 @@ int amdgpu_bo_cpu_map(amdgpu_bo_handle bo, void **cpu)
 		pthread_mutex_unlock(&bo->cpu_access_mutex);
 		return -errno;
 	}
-	amdgpu_add_handle_to_table(bo);
 	bo->cpu_ptr = ptr;
 	bo->cpu_map_count = 1;
 	pthread_mutex_unlock(&bo->cpu_access_mutex);
+	amdgpu_add_handle_to_table(bo);
 
 	*cpu = ptr;
 	return 0;
-- 
2.7.4

